### PR TITLE
feat: Handle ⇧⌘S shortcut and the Save as... Menu Item in the File Menu (SDKCF-5137)

### PR DESCRIPTION
**Pull request by malayli**

### CODE REVIEW

File: PusherMainView/PusherMainViewTests/PusherStoreSpec.swift [hunk 1]
```diff
@@ -63,6 +69,8 @@ extension ObserverMock: PusherInteractable {
 
 final class PusherStoreSpec: QuickSpec {
     override func spec() {
+        var router = RouterMock()
+        let viewController = NSViewController()
         let fileName = "pusherfile.txt"
         let filePath = FileManager.default.currentDirectoryPath.appending("/\(fileName)")
```
Comment (imaginaris) [inline PusherMainView/PusherMainViewTests/PusherStoreSpec.swift:75]: Can we use temp directory here?

File: PusherMainView/PusherMainView/PusherStore.swift [hunk 2]
```diff
@@ -248,6 +248,13 @@ extension PusherStore: PusherInteracting {
                 reduce(result: .failure(error, actionType))
             }
 
+        case .saveFileAs(let text, let fromViewController, let completion):
+            router.saveFileAs(from: fromViewController) { fileURL in
+                self.dispatch(actionType: .chooseFile)
```
Comment (imaginaris) [inline PusherMainView/PusherMainView/PusherStore.swift:253]: What does this action do?

Comment (malayli) [reply PusherMainView/PusherMainView/PusherStore.swift]: @imaginaris I renamed `chooseFile` to `enableSaveMenuItem `. Clearer I think.

File: PusherMainView/PusherMainViewTests/PusherStoreSpec.swift (hunk 1 above)
Comment (malayli) [reply PusherMainView/PusherMainViewTests/PusherStoreSpec.swift:75]: @imaginaris Can you create a ticket for this improvement?
It's not part of this PR.

Comment (imaginaris) [reply PusherMainView/PusherMainViewTests/PusherStoreSpec.swift:75]: Ok.  Please check SDKCF-5520

File: PusherMainView/PusherMainView/PusherStore.swift (hunk 2 above)
Comment (imaginaris) [reply PusherMainView/PusherMainView/PusherStore.swift]: Yeah. Much clearer 🙂

**Review by imaginaris — APPROVED** [review]

File: PusherMainView/PusherMainViewTests/PusherStoreSpec.swift (hunk 1 above)
Comment (imaginaris) [reply PusherMainView/PusherMainViewTests/PusherStoreSpec.swift:75]: Sorry @malayli please ignore. I just noticed that "pusherfile.txt" is being removed after each test. There's no need to use temp directory.

**Pull request merged**